spec/importer_spec.rb

```diff
@@ -0,0 +1,191 @@
+require 'spec_helper'
+require 'ronin/masscan/importer'
+require 'masscan/output_file'
+require 'ronin/db'
+
+RSpec.describe Ronin::Masscan::Importer do
+  let(:masscan_path) { File.expand_path(File.join(__dir__, '..', 'spec', 'fixtures', 'masscan.json')) }
+  let(:output_file)  { Masscan::OutputFile.new(masscan_path) }
+
+  before(:all) do
+    Ronin::DB.connect('sqlite3::memory:')
+  end
+
+  after(:all) do
+    Ronin::DB::IPAddress.destroy_all
+    # Ronin::DB::Port.destroy_all
+    Ronin::DB::OpenPort.destroy_all
+  end
+
+  describe '.import_file' do
+    let(:expected_values) do
+      [
+        Ronin::DB::IPAddress,
+        Ronin::DB::Port,
+        Ronin::DB::OpenPort,
+        Ronin::DB::IPAddress,
+        Ronin::DB::Port,
+        Ronin::DB::OpenPort,
+        Ronin::DB::IPAddress
+      ]
+    end
+
+    it 'must import records from file' do
+      yielded_values = []
+
+      subject.import_file(masscan_path) do |value|
+        yielded_values << value
+      end
+
+      expect(yielded_values.size).to eq(7)
+      expect(yielded_values.map(&:class)).to match(expected_values)
+    end
+  end
+
+  describe '.import' do
+    let(:expected_values) do
+      [
+        Ronin::DB::IPAddress,
+        Ronin::DB::Port,
+        Ronin::DB::OpenPort,
+        Ronin::DB::IPAddress,
+        Ronin::DB::Port,
+        Ronin::DB::OpenPort,
+        Ronin::DB::IPAddress
+      ]
+    end
+
+    context 'when an unknown record type is encountered' do
+      before do
+        allow(output_file).to receive(:each).and_yield("unknown")
+      end
+
+      it 'must raise NotImplementedError' do
+        expect { |b|
+          subject.import(output_file, &b)
+        }.to raise_error(NotImplementedError)
+      end
+    end
+
+    context 'when a block is given' do
+      it 'must yield imported records' do
+        yielded_values = []
+
+        subject.import_file(masscan_path) do |value|
+          yielded_values << value
+        end
+
+        expect(yielded_values.size).to eq(7)
+        expect(yielded_values.map(&:class)).to match(expected_values)
+      end
+    end
+
+    context 'when no block is given' do
+      it 'must return imported records' do
+        expect(subject.import(output_file).size).to eq(7)
+        expect(subject.import(output_file).map(&:class)).to match(expected_values)
+      end
```

Comment:
I'd avoid calling `.import` twice, and instead save it to a local variable like `imported_values`, then test those. You could also probably copy/paste some of the `expect()` logic from the specs below and re-use them here.